nni/compression/experiment/config/compression.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Type, Union
+
+from torch.nn import Module
+
+from nni.experiment.config.base import ConfigBase
+from .pruner import PrunerConfig
+from .quantizer import QuantizerConfig
+from .common import ComparableType
+
+__all__ = ['CompressionConfig']
+
+
+@dataclass
+class CompressionConfig(ConfigBase):
+    # constraints
+    params: Union[str, int, float, List[Union[str, int, float]], None] = None
+    flops: Union[str, int, float, List[Union[str, int, float]], None] = None
+    # latency: float, List[float], None
+    metric: Union[ComparableType, List[ComparableType], None] = None
```

Comment:
Can it be string, list, tuple, etc? Or it must be a number?

Reply:
any class that implements `__lt__`

Reply:
Can you give a example where metric is not trivial type?

Reply:
some evaluator may output `{'acc': xxx, 'f1': xxx}`, in this case we don't know how to compare the metric. So if user want to set metric as constrain, he need to define a `comparable dict`.

Also this is a future feature, only support the evaluator that return `number` now...

Reply:
only support `float` right now.